train/net_utils.py

```diff
@@ -134,7 +142,7 @@ def create_convnetv1(inputs, residual_block_num, residual_filter_num, moves_num,
     # Value head
     flow_val = conv_block(flow,
                           filter_size=1,
-                          output_channels=2,
+                          output_channels=value_head_conv_output_channels_num,

```

Comment:
1. If the input to the layer is a 2d n×m matrix, a single convolution of size 3x3 (for example) can be applied, creating an output with size n×m. The 9 weights of the conv kernel are trained by the network. Multiple channels means applying k different 3×3 convolutions, each with different 9 parameters, resulting in output with shape n×m×k. In case the input is 3d, n×m×c, which may be an output of conv layer with multiple channels (or the raw input in our case), conv layer with k channels means k different convolutions, each with 3d kernel 3×3×c, resulting in output with shape n×m×k.

2. I dont have much intuition about it either... Except some general ideas of nets and layers which we can talk about live. Other than the batch norm, its just conv and dense layers, which is quite simple, but im sure there is a deeper understanding of it